management/email_administrator.py

```diff
@@ -31,6 +31,9 @@
 if content == "":
 	sys.exit(0)
 
+if (content.find('error:') != -1):
+  subject = 'Error ' + subject
```

Comment:
The rest of the python file uses string formatting with %s so this should probably be kept constant for consistency (and I would also add a comment just in case it is not clear (!) what is being done):

```
# check if the content contains an error message
if (content.find('error:') != -1):
  subject = 'Error ' + subject
```

Overall this fixes an irritating bug, good job!

Reply:
@ctrl-i, thanks for the review!  Really appreciate it.

> string formatting with %s

Ah, I was using the same `+` syntax as line 25.  I feel like that's more readable, if that's okay.

> I would also add a comment

Good call, I just added one.

Reply:
> Ah, I was using the same `+` syntax as line 25. I feel like that's more readable, if that's okay.

Yep, you're right.  Keep it with + as it is more readable and already used.

I had a problem overnight with this... as email_administrator.py is used to send several emails from different scripts I had another email not related to TLS Provisioning append Error to the subject line as its content contained "error:".  I think we therefore need to tighten the text we are trying to find or double check that the subject line is for the TLS Provisioning mail (or possibly both).

Reply:
@ctrl-i, thanks for trying it live, that was smart.  Could you copy-paste the contents of that email here?  I'll figure something out based on that.

Reply:
For privacy reasons I cannot do that.

Reply:
@ctrl-i, no harm in redacting whatever you don't feel comfortable including.  It would just be helpful to understand what we're dealing with – the general nature and structure of the error message.

Reply:
@ctrl-i, given that conditional error text isn't so simple, I abandoned that idea in favor of simply changing the email subject line from "Error Provisioning TLS Certificate" to "TLS
Certificate Provisioning Results".

Reply:
Sometimes the best fixes are the easiest ones :)